src/lib.rs

```diff
@@ -340,4 +340,25 @@ where
             }
         }
     }
+
+    #[cfg(feature = "rng")]
+    /// If supported, generate a 32 bit random value.
+    ///
+    /// Warning: prepare_for_xxx() MUST be called after this operation to set modulation and packet parameters (for example: xxx = tx, rx, cad).
+    /// Do not set modulation and packet parameters, do a random number generation, then initiate Tx, Rx, or CAD.
```

Comment:
Would it make sense to take note of the modulation/packet parameters before the function starts messing with stuff, and resetting everything like it was when the random number has been generated? Seems to me like it would be a lot less error-prone. Not sure if it's actually possible though.

Reply:
Or, if that's not possible, only accept RNG requests when the chip is in a certain operating mode. Ie, `Standby`?

Reply:
Fair enough, let's not overcomplicate anything for now. Yes, I removed the RngCore implementation from #9. For now that PR is effectively just a mirror of #10, because I intend for all the RngCore functionality to be handled in lowaran-device (if it's accepted by the devs over there). I think this PR should actually be merged first, so that the new RNG API is made available to lorawan-device. If we need to come back to lora-phy for an RngCore implementation, it can be done in #9, otherwise I'll just end up closing it.

If we end up taking that direction, though, I wonder if we would benefit from having a custom async RNG trait defined in this crate, rather than just hiding the get_random method behind a feature gate. Thoughts?